pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -768,6 +819,89 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    pub(crate) async fn delete_all(self: &Arc<Self>) -> anyhow::Result<()> {
+        let mut receiver = {
+            let mut locked = self.upload_queue.lock().unwrap();
+            let stopped = match &mut *locked {
+                UploadQueue::Uninitialized => {
+                    anyhow::bail!("upload queue is not stopped but initialized")
+                }
+                UploadQueue::Initialized(_) => {
+                    anyhow::bail!("upload queue is not stopped but initialized")
+                }
+                UploadQueue::Stopped(stopped) => stopped,
+            };
+
+            if !matches!(stopped.deleted_at, SetDeletedFlagProgress::Successful(_)) {
+                anyhow::bail!("deleted_at is not set")
+            }
+
+            debug_assert!(stopped
+                .upload_queue_for_deletion
+                .inprogress_tasks
+                .is_empty());
+            debug_assert!(stopped
+                .upload_queue_for_deletion
+                .queued_operations
+                .is_empty());
+
```

Comment:
Consider: debug_assert!(stopped.upload_queue_for_deletion.no_pending_work())

and hide the various options for "pending_work" within the upload_queue

Reply:
Done